app_server/app/policies/Authenticator.js

```diff
@@ -190,8 +190,15 @@ Class.validateAccount = function (server, session) {
       return null;
     }
 
-    return session.username === cached.username &&
-           session.password === cached.password;
+    var isAccountValidated = session.username === cached.username &&
+                             session.password === cached.password;
+
+    if (!isAdminScope(session.scope)) {
+      return isAccountValidated;
+    } else {
+      return isAccountValidated && session.csrfToken &&
+             session.csrfToken === cached.encryptedPassword;
```

Comment:
a CSRF token is used, to check "a form data posted by a user" against the "crsf token in the headers", to make sure that every post request is originated from our site.